### ISSUE TITLE
Support for Nested Subcommands?

### ISSUE THREAD

**Issue by alecbcs** [DataDrake/cli-ng#12]
Hey @DataDrake,

On one of the projects I'm working on I'd kinda love to be able to nest subcommands. For example,

```
hyprspace peers add

hyprspace peers remove
```

Where `hyprspace` is the name of the program, `peers` is the top level subcommand, and `add/remove` are children of `peers`.

Looking around I didn't see an obvious way to do this in `cli-ng` but, perhaps I'm just not looking in the right place? If this isn't already implemented, would this be something you are interested in?

**Comment by DataDrake** [comment]
Ironically, this was relatively easy to do in cli-ng v1 because of the API. I'll need to rethink how I did that and get back to you about changing it at some point. Preferably without going to a v3 API.

**Comment by vsoch** [comment]
hey @DataDrake just wanted to say +1 I'd really appreciate this as well! I use cli-ng for a bunch of my libraries, but I'm hitting the point where needing subcommands is going to require me to switch to a different one.

**Comment by DataDrake** [comment]
Yep. I'll dig into this again for cli-ng v3. It will affect a bunch of things, so I'll need to pay a bit for attention. On the plus side, Single mode will allow sub-commands when I'm done.
